lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2023,6 +2036,83 @@ quitStakePool ctx (ApiT wid) body = do
 
     genChange = delegationAddress @n
 
+listStakeKeys
+    :: forall ctx s n k.
+        ( ctx ~ ApiLayer s k
+        , s ~ SeqState n k
+        , HasNetworkLayer IO ctx
+        , Typeable n
+        , Typeable s
+        )
+    => (Address -> Maybe RewardAccount)
+    -> ctx
+    -> ApiT WalletId
+    -> Handler (ApiStakeKeys n)
+listStakeKeys stakeRef ctx (ApiT wid) = do
```

Comment:
This is quite a big function. Is it at all possible to break it up a little? Then it might be possible to unit test some of it.

Reply:
👍 The thought occurred to me, but wasn't sure if worth it; thanks for suggesting, will give it a shot

Reply:
I split off a helper that could run in `Identity`. But to actually unit-test it, I can't get a way with only defining Api types like I do here. Will do in a separate PR.